cmd/admin-config-set.go

```diff
@@ -0,0 +1,128 @@
+/*
+ * Minio Client (C) 2017 Minio, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cmd
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+
+	"github.com/fatih/color"
+	"github.com/minio/cli"
+	"github.com/minio/mc/pkg/console"
+	"github.com/minio/minio/pkg/madmin"
+	"github.com/minio/minio/pkg/probe"
+)
+
+var adminConfigSetCmd = cli.Command{
+	Name:   "set",
+	Usage:  "Set new config file to a Minio server/cluster.",
+	Before: setGlobalsFromContext,
+	Action: mainAdminConfigSet,
+	Flags:  globalFlags,
+	CustomHelpTemplate: `NAME:
+  {{.HelpName}} - {{.Usage}}
+
+USAGE:
+  {{.HelpName}} ALIAS/
+
+FLAGS:
+  {{range .VisibleFlags}}{{.}}
+  {{end}}
+EXAMPLES:
+  1. Set server configuration of a Minio server/cluster.
+     $ cat myconfig | {{.HelpName}} myminio/
+
+`,
+}
+
+// configSetMessage container to hold locks information.
+type configSetMessage struct {
+	Status               string               `json:"status"`
+	SetConfigStatus      bool                 `json:"-"`
+	SetConfigNodeSummary []madmin.NodeSummary `json:"nodesSummary"`
+}
+
+// String colorized service status message.
+func (u configSetMessage) String() (msg string) {
+	defer func() {
+		msg = console.Colorize("SetConfig", msg)
+	}()
+	if u.SetConfigStatus {
+		msg += "Setting new configuration file has been successful.\n"
+	} else {
+		msg += "Setting new configuration file has failed.\n"
+	}
+	for _, s := range u.SetConfigNodeSummary {
+		msg += fmt.Sprintf("  %s.. ", s.Name)
```

Comment:
Actually here the output would be nicer to be like - success is printed at the end based on the errors..

```
$ mc admin config set myminiold1/ < config.json
New configuration applied to `172.17.0.2:9000`  successfully.
New configuration applied to `172.17.0.3:9000`  successfully.
New configuration applied to `172.17.0.4:9000`  successfully.
New configuration applied to `172.17.0.5:9000`  successfully.
Setting new Minio configuration has been successful.
```